Emby.Server.Implementations/AppBase/BaseConfigurationManager.cs

```diff
@@ -19,11 +20,44 @@ namespace Emby.Server.Implementations.AppBase
     /// </summary>
     public abstract class BaseConfigurationManager : IConfigurationManager
     {
+        private readonly IFileSystem _fileSystem;
+
+        private readonly ConcurrentDictionary<string, object> _configurations = new ConcurrentDictionary<string, object>();
+
+        private ConfigurationStore[] _configurationStores = Array.Empty<ConfigurationStore>();
+        private IConfigurationFactory[] _configurationFactories = Array.Empty<IConfigurationFactory>();
+
         /// <summary>
-        /// Gets the type of the configuration.
+        /// The _configuration loaded.
         /// </summary>
-        /// <value>The type of the configuration.</value>
-        protected abstract Type ConfigurationType { get; }
+        private bool _configurationLoaded;
+
+        /// <summary>
+        /// The _configuration sync lock.
+        /// </summary>
+        private object _configurationSyncLock = new object();
```

Comment:
There are a lot of comments like this one (in general, not this specific pull request) that I would consider useless, but it's not a major issue.